src/openms/source/ANALYSIS/TARGETED/MetaboTargetedAssay.cpp

```diff
@@ -0,0 +1,498 @@
+              adduct = min_distance_feature->getPeptideIdentifications()[0].getHits()[j].getMetaValue("modifications");
+
+              // change format of description [name] to name
+              description.erase(remove_if(begin(description), end(description), [](char c) { return c == '[' || c == ']'; }), end(description));
+
+              // change format of adduct information M+H;1+ -> [M+H]1+
+              String adduct_prefix = adduct.prefix(';').trim();
+              String adduct_suffix = adduct.suffix(';').trim();
+              adduct = "[" + adduct_prefix + "]" + adduct_suffix;
+
+              min_id_mz_error = abs(current_id_mz_error);
+             }
+          }
+          // use the identification with the lowest mass deviation
+          v_description.push_back(description);
+          v_sumformula.push_back(sumformula);
+          v_adduct.push_back(adduct);
+        }
+        else
+        {
+          // count UNKNOWN via transition group counter
+          v_description.push_back(String(description + "_" + transition_group_counter));
+          v_sumformula.push_back(sumformula);
+          v_adduct.push_back(adduct);
+        }
+
+      double highest_precursor_mz = 0.0;
+      float highest_precursor_int = 0.0;
+      int highest_precursor_charge = 0;
+      MSSpectrum highest_precursor_int_spectrum;
+      MSSpectrum transition_spectrum;
+      String native_id;
+
+      // find precursor/spectrum with highest intensity precursor
+      vector <size_t> index = it->second;
+
+      for (auto index_it = index.begin(); index_it != index.end(); ++index_it)
+      {
+        const MSSpectrum &spectrum = spectra[*index_it];
+
+        // check if MS2 spectrum is empty
+        if (spectrum.empty())
+        {
+          LOG_WARN << "Empty MSMS spectrum was provided. Please hava a look. Index: " << *index_it << std::endl;
+          continue;
+        }
+
+        const vector <Precursor> &precursor = spectrum.getPrecursors();
+
+        // get m/z and intensity of precursor
+        // only spectra with precursors are in the map, therefore no need to check for their presence
+        double precursor_mz = precursor[0].getMZ();
+        float precursor_int = precursor[0].getIntensity();
+        int precursor_charge = precursor[0].getCharge();
+
+        native_id = spectrum.getNativeID();
+
+        // spectrum with highest intensity precursor
+        if (precursor_int > highest_precursor_int)
+        {
+          highest_precursor_int = precursor_int;
+          highest_precursor_mz = precursor_mz;
+          highest_precursor_int_spectrum = spectrum;
+          highest_precursor_charge = precursor_charge;
+        }
+          transition_spectrum = highest_precursor_int_spectrum;
+      }
+
+      // if only one MS2 is available and the consensus method is used - jump right to the transition list calculation
+      // fallback: highest intensity precursor
```

Comment:
I dont think this comments reflects well what happens below, can you adjust that? It seems you use the highest intensity spectrum, then find otehr spectra with high enough cosine angle and add all those together?

Reply:
Yes, you are right.
```
      // check if more than one MS/MS spectrum is available to use the consensus method
      // the MS/MS spectrum of the highest intensity precursor is used as reference and compared
      // to the other MS/MS spectrum of a specific feature.
      // if the cosine similarity is over the manually set threshold these are merged via SpectraMerger
```